### ISSUE TITLE
[iOS] Time input is unpredictable

### ISSUE THREAD

**Issue by roxteddy** [capacitor-community/date-picker#25]
**Describe the bug**
In time mode, datepicker relies on phone settings rather that "format" option

**To Reproduce**
1. Disable 24h mode in device settings
2. open datepicker with mode: time, format: "HH:mm", date: "14:55" 
4. error "failed to parse date"

**Expected behavior**
It should rely on the "format" option

**Smartphone (please complete the following information):**
 - Device: iPad 6
 - OS: iOS 14.1

**Comment by roxteddy** [comment]
according to the documentation : https://developer.apple.com/documentation/foundation/dateformatter
chapter : Working With Fixed Format Date Representations

we should add `RFC3339DateFormatter.locale = Locale(identifier: "en_US_POSIX")` to the code in order to make the dateformat works

However I have to test further because I has other problems then.
It seems to be a lot of localization problems here and there. I started my own fork to investigate. Hope you guys will get back here somedays so we can discuss this ;)

**Comment by danielprrazevedo** [comment]
Hello @roxteddy, sorry for the delay in responding. I had turbulent weeks.
I agree that, especially in IOS, we have many problems in formatting dates and one of the reasons for this is that I am very willing, but my swift is still not very good.
If you want to investigate this specific problem, I will be happy to accept the PR. It is in my plans to redo this part of the plugin for IOS, but I haven't had time yet, I believe that in December I can fix some things.
One thing I'm starting to rethink if it's worth keeping is the is24 selector, it caused a lot of problems and confusion. I wonder what do you think of that?

**Comment by roxteddy** [comment]
Hello @danielprrazevedo 

The is24h selector, it definitely needs a rework. I talked about my idea in the other issue.

For the current issue, I don't have problem anymore with my device set to country : France, neither with device setting 24h mode on or off.
However, now I meet the same problem with the device set to country : United States with 24h mode on.
It fails parsing the date but even stranger, my debugger goes thru line 
`self.call?.reject("Failed to parse date")`
but I don't get the rejection on typescript side.
So the picker instantly dismiss and I get no error.

I don't know why the rejection not happening but the line I gave before from Apple documentation fixes the issue. I'll make a PR for it
